frontend/packages/topology/src/utils/element-utils.ts

```diff
@@ -29,16 +29,50 @@ const leafNodeElements = (nodeElements: Node | Node[] | null): Node[] => {
     return nodes;
   }
 
-  const children: GraphElement[] = nodeElements.getChildren();
-  if (_.size(children)) {
+  if (nodeElements.isGroup()) {
     const leafNodes: Node[] = [];
-    _.forEach(children.filter((e) => isNode(e)), (element: Node) => {
-      leafNodes.push(...leafNodeElements(element));
-    });
+    const children: GraphElement[] = nodeElements.getChildren();
+    if (_.size(children)) {
+      _.forEach(children.filter((e) => isNode(e)), (element: Node) => {
+        leafNodes.push(...leafNodeElements(element));
+      });
+    }
     return leafNodes;
   }
 
   return [nodeElements];
 };
 
-export { groupNodeElements, leafNodeElements };
+const getElementPadding = (element: GraphElement): number => {
+  const stylePadding = element.getStyle<GroupStyle>().padding;
+  if (!stylePadding) {
+    return 0;
+  }
+
+  if (Array.isArray(stylePadding)) {
+    // For a padding that is not consistent on all sides, use the average padding
```

Comment:
Might be more appropriate to use the max padding? Otherwise there's more potential for overlap?